heron/common/src/python/pex_loader.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2016 Twitter. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''pex_loader.py: module for dynamically loading pex'''
+
+import os
+import re
+import sys
+import zipimport
+import zipfile
+
+from heron.common.src.python.log import Log
+
+egg_regex = r"^(\.deps\/[^\/\s]*\.(egg|whl))\/"
+
+def _get_deps_list(abs_path_to_pex):
+  """Get a list of paths to included dependencies in the specified pex file
+
+  Note that dependencies are located under `.deps` directory
+  """
+  pex = zipfile.ZipFile(abs_path_to_pex, mode='r')
+  deps = list(set([re.match(egg_regex, i).group(1) for i in pex.namelist()
+                   if re.match(egg_regex, i) is not None]))
+  return deps
+
+def load_pex(path_to_pex, include_deps=True):
+  """Loads pex file and its dependencies to the current python path"""
+  abs_path_to_pex = os.path.abspath(path_to_pex)
+  Log.debug("Add a pex to the path: " + abs_path_to_pex)
+  sys.path.insert(0, abs_path_to_pex)
+
+  # add dependencies to path
+  if include_deps:
+    for dep in _get_deps_list(abs_path_to_pex):
+      Log.debug("Add a new dependency to the path: " + dep)
+      sys.path.insert(0, os.path.join(abs_path_to_pex, dep))
+
+  Log.debug("Python path: " + str(sys.path))
+
+def resolve_heron_suffix_issue(abs_pex_path, class_path):
+  """Resolves duplicate package suffix problems
+
```

Comment:
I was wondering - instead of starting with the name 'heron' - can we start with some scrambled version such as "A89109120120" - a UUID and the likelihood of it clashing is very very less, isn't? Just an idea - not sure how feasible it is?